client/my-sites/domains/domain-management/list/all-domains.jsx

```diff
@@ -330,19 +327,31 @@ class AllDomains extends Component {
 				initialSortOrder: -1,
 				sortFunctions: [
 					( first, second, sortOrder ) => {
-						const { listStatusWeight: firstStatusWeight } = resolveDomainStatus( first, null, {
-							getMappingErrors: true,
-						} );
-						const { listStatusWeight: secondStatusWeight } = resolveDomainStatus( second, null, {
-							getMappingErrors: true,
-						} );
+						const { listStatusWeight: firstStatusWeight } = resolveDomainStatus(
+							first,
+							null,
+							translate,
+							dispatch,
+							{
+								getMappingErrors: true,
+							}
+						);
+						const { listStatusWeight: secondStatusWeight } = resolveDomainStatus(
```

Comment:
Using `this.props.dispatch()` directly is not ideal, but looks like an acceptable compromise in this scenario.

Reply:
`this.props.dispatch` is surely unusual when used together with `connect`, but on the other hand, it's very close to how you would use `useDispatch` in a functional component with hooks. It think there's nothing wrong using it that way.